src/client/main.c

```diff
@@ -0,0 +1,71 @@
+#include <errno.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "client.h"
+#include "libhirte/common/common.h"
+#include "libhirte/common/opt.h"
+
+const struct option options[] = { { ARG_HELP, no_argument, 0, ARG_HELP_SHORT }, { NULL, 0, 0, '\0' } };
+
+#define OPTIONS_STR ARG_HELP_SHORT_S
+
+static char *op;
+static char **opargv;
+static int opargc;
+static bool no_action = false;
+
+static void usage(char *argv[]) {
+        fprintf(stderr, "Usage: %s COMMAND ... \n", argv[0]);
+}
+
+static int get_opts(int argc, char *argv[]) {
+        int opt = 0;
+
+        while ((opt = getopt_long(argc, argv, OPTIONS_STR, options, NULL)) != -1) {
+                if (opt == ARG_HELP_SHORT) {
+                        usage(argv);
+                        no_action = true;
+                        return 0;
+                } else {
+                        fprintf(stderr, "Unsupported option %c\n", opt);
+                        usage(argv);
+                        return -EINVAL;
+                }
+        }
+
+        if (optind < argc) {
+                op = argv[optind++];
+                opargv = &argv[optind];
+                opargc = argc - optind;
+        } else {
+                fprintf(stderr, "No command given\n");
+                usage(argv);
+                return -EINVAL;
+        }
+
+        return 0;
```

Comment:
```suggestion
        if (optind < argc) {
                op = argv[optind++];
                opargv = &argv[optind];
                opargc = argc - optind;
                return 0;
        }
        fprintf(stderr, "No command given\n");
        usage(argv);
        return -EINVAL;
```

Reply:
As we discussed, I think that a case like this should be in else clause, and not outside, as if this is the main flow.

Reply:
What do you mean by main flow? `else` or not, both are semantically the same. The only difference is the early return in the suggested code. 
Again, like the other comment this is not a blocker.